### ISSUE TITLE
Unpack dictionary as class keyword arguments

### ISSUE THREAD

**Issue by BrianPugh** [microsoft/pyright#3469]
**Describe the bug**
Pyright doesn't like unpacking a dictionary as class keyword arguments.

**To Reproduce**

```python
kwargs = {"foo": 5}


# [Pyright reportGeneralTypeIssues] [E] Expected class type but received "dict[str, int]"
class Foo(**kwargs):
    pass
```

**Expected behavior**
It should be fine, same as when keyword arguments are directly assigned.

**Screenshots or Code**
My exact situations involves a metaclass; this situation below is a simplified version.
```python
class MyMeta(type):
    def __contains__(self, val):
        return val in [1, 2, 3]


class MyMetaHelper(metaclass=MyMeta):
    pass


class MyClassGood1(MyMetaHelper):
    pass


assert 2 in MyClassGood1


class MyClassGood2(MyMetaHelper, foo=5):
    pass


assert 2 in MyClassGood2


kwargs = {"foo": 5}


# [Pyright reportGeneralTypeIssues] [E] Expected class type but received "dict[str, int]"
class MyClassBad(MyMetaHelper, **kwargs):
    pass


# [Pyright reportGeneralTypeIssues] [E] Operator "in" not supported for types "Literal[2]" and "Type[MyClassBad]"
assert 2 in MyClassBad
```

**Comment by erictraut** [comment]
Thanks for the bug report. A similar problem also exists for `*args` used in the same context. I've addressed both issues. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.248, which I just published. It will also be included in the next release of pylance.
